src/erlfmt_cli.erl

```diff
@@ -83,18 +93,48 @@ do_unprotected(Opts, Name) ->
             erlang:halt(2)
     end.
 
+-dialyzer({no_improper_lists, [read_stdin/1]}).
+
+read_stdin(Data) ->
+    case io:get_chars(standard_io, "", 4096) of
+        MoreData when is_binary(MoreData) -> read_stdin([Data | MoreData]);
+        eof -> {ok, Data};
+        {error, Reason} -> throw({error, Reason})
+    end.
+
 format_file(FileName, Config) ->
-    #config{pragma = Pragma, width = Width, verbose = Verbose, out = Out} = Config,
+    #config{pragma = Pragma, width = Width, verbose = Verbose, check = Check, out = Out} = Config,
     case Verbose of
         true -> io:format(standard_error, "Formatting ~s\n", [FileName]);
         false -> ok
     end,
     Options = [{pragma, Pragma}] ++ [{width, Width} || Width =/= undefined],
-    case erlfmt:format_file(FileName, Out, Options) of
+    Result =
+        case Check of
+            true ->
+                {ok, OriginalBin} =
+                    case FileName of
+                        stdin -> read_stdin([]);
+                        _ -> file:read_file(FileName)
+                    end,
+                Original = unicode:characters_to_list(OriginalBin),
+                case erlfmt:format_string(Original, Options) of
+                    {ok, Formatted, FormatWarnings} ->
+                        case Formatted of
+                            Original -> {ok, FormatWarnings};
+                            _ -> {warn, FormatWarnings}
+                        end;
+                    Other ->
+                        Other
+                end;
```

Comment:
Let's extract this to a function

Reply:
Also, does this compose well with `require-pragma`?

Reply:
Great idea, I extracted it.
I also added a test for require-pragma